### PR TITLE
docs(Theme): update Theme docs with new guidance/usage

### PR DESCRIPTION
**Pull request by dakahn**
partially closes [#11293](https://github.com/carbon-design-system/carbon/issues/11293)

Updates the /code page for Themes with new usage/guidance clarity based on some support work we've done

### CODE REVIEW

File: src/pages/guidelines/themes/code.mdx [hunk 1]
```diff
@@ -49,10 +48,18 @@ yarn add @carbon/themes
...
-your project. By default, `@carbon/themes` provides a set of color tokens that
-are pre-defined for a specific theme. Currently, we offer the following color
-themes: white, gray 10, gray 90, gray 100 .
+There are three ways to utilize Carbon Themes. First, and maybe the simplest way
+is to bring in our
+[`<Theme>`](https://react.carbondesignsystem.com/?path=/story/components-theme--default)
+component and pass your desired theme as a prop. Second, if you need more
+granular control of what gets themed and when and you're using `@carbon/styles`
+or `@carbon/react` you already have access to sass theming with no need for an
+extra package. Thirdly, can use `@carbon/themes` in JavaScript or Sass by
```
Comment (aagonzales) [inline src/pages/guidelines/themes/code.mdx:57]:
```suggestion
extra package. Thirdly, you can use `@carbon/themes` in JavaScript or Sass by
```

File: src/pages/guidelines/themes/code.mdx [hunk 2]
```diff
@@ -49,10 +48,18 @@ yarn add @carbon/themes
...
 ## Usage
 
-You can use `@carbon/themes` in JavaScript or Sass by including this package in
-your project. By default, `@carbon/themes` provides a set of color tokens that
-are pre-defined for a specific theme. Currently, we offer the following color
-themes: white, gray 10, gray 90, gray 100 .
+There are three ways to utilize Carbon Themes. First, and maybe the simplest way
+is to bring in our
+[`<Theme>`](https://react.carbondesignsystem.com/?path=/story/components-theme--default)
+component and pass your desired theme as a prop. Second, if you need more
```
Comment (aagonzales) [inline src/pages/guidelines/themes/code.mdx:54]: The second method description was a little confusing to me.

File: src/pages/guidelines/themes/code.mdx [hunk 3]
```diff
@@ -49,10 +48,18 @@ yarn add @carbon/themes
...
+or `@carbon/react` you already have access to sass theming with no need for an
+extra package. Thirdly, can use `@carbon/themes` in JavaScript or Sass by
+including this package in your project.
+
+By default, `@carbon/themes` provides a set of color tokens that are pre-defined
+for a specific theme. Currently, we offer the following color themes: white,
+gray 10, gray 90, gray 100 .
 
 You can preview all of the token values for this on the
 [Carbon Design System website](https://www.carbondesignsystem.com/guidelines/color/usage)
```
Comment (aagonzales) [inline src/pages/guidelines/themes/code.mdx:65]: Extra change ... they're already on the website for this is so you don't need to say the website. 
```suggestion
[color guidelines(https://www.carbondesignsystem.com/guidelines/color/usage) page.
```

**Review by tw15egan — APPROVED** [review]

File: src/pages/guidelines/themes/code.mdx [hunk 4]
```diff
@@ -87,7 +94,7 @@ Alternatively, you can set the global theme variable then call the mixin without
 passing in a theme name.
 
 ```scss
-@import '@carbon/themes/scss/themes';
+@use '@carbon/themes/scss/themes';
 
 $theme: $theme--g10;
```
Comment (tay1orjones) [inline src/pages/guidelines/themes/code.mdx:99]: This usage is outdated, right?

**Review by tay1orjones — CHANGES_REQUESTED** [review]
This will need the same feedback from https://github.com/carbon-design-system/carbon/pull/11635, I think.

Like using both modules
```
@use '@carbon/themes/scss/themes';
@use '@carbon/themes/scss/theme';
```

and then making sure the namespace is there when calling the mixin
```
@include theme.theme(themes.$g90)
```

Also the $theme--g## variables etc are outdated and need updated to their new counterparts in the example with `.my-dark-theme` and `.my-darker-theme`. I can't comment on those lines in the PR review since they weren't changed.

File: src/pages/guidelines/themes/code.mdx (hunk 4 above)
Comment (dakahn) [reply src/pages/guidelines/themes/code.mdx:99]: Is it? I thought that's how you accessed the theming tokens, but I'd bet i'm definitely wrong

Comment (dakahn) [reply src/pages/guidelines/themes/code.mdx:99]: i thought that was how we still brought in themes for custom work? but I could just be doing it wrong

**Review by aagonzales — APPROVED** [review]

**Comment by tay1orjones** [comment]
I've totally overhauled this one with a bunch of theme related changes. Please re-review and let me know what you think.

**Review by tw15egan — APPROVED** [review]

**Comment by tay1orjones** [comment]
@tw15egan No! Good distinction. In this case it means that instead of having `@carbon/themes` installed and doing:

```scss
@use '@carbon/themes/scss/theme';
```

You can instead access this stuff from `@carbon/react`
```scss
@use '@carbon/react/scss/theme';
```

This is limited to _just the styles_ as well. If someone's wanting to get the exports for usage  in javascript, they'll need to install the `@carbon/themes` package.

I'll update this verbiage to say "you likely don't need to install the themes package" instead of "there's no reason to install the themes package"

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**